clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1005,8 +1003,22 @@ public void assign(Collection<TopicPartition> partitions) {
         // send any new fetches (won't resend pending fetches)
         fetcher.sendFetches();
 
+        // if no fetches could be sent at the moment (which can happen if a partition leader is in the
+        // blackout period following a disconnect), then we only block for the retry backoff duration.
+        if (!fetcher.hasInFlightFetches())
+            timeout = retryBackoffMs;
```

Comment:
Should we do the following to be safe?

```java
timeout = Math.min(retryBackoffMs, timeout)
```

Also, could `retryBackoffMs` be too conservative (i.e. is it possible for us to be nearer to the end of the blackout period)? Does it matter?